File: 0997-find-the-town-judge/0997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        if not trust:
            return n if n == 1 else -1
    
        trustCount = {}
        posJudge = None
        
        haveTrust = set()
        
        for a, b in trust:
            haveTrust.add(a)          
                
            if b not in trustCount:
                trustCount[b] = 0
            
            trustCount[b] += 1
        
        for key, value in trustCount.items():
            if value == n - 1 and key not in haveTrust:
                return key
            
        return -1
        
        

File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        unordered_set<int> hasTrust;
        unordered_map<int, int> trustCount;
        
        for (int i = 1; i <= n; i++) {
            hasTrust.insert(i);
        }
        
        for (auto relation: trust) {
            int a = relation[0];
            int b = relation[1];
            
            if (hasTrust.count(a)) {
                hasTrust.erase(a);
            }
            trustCount[b]++;
        }
        
        for (auto maybeJudge: hasTrust) {
            if (trustCount[maybeJudge] != n - 1) {
                continue;
            }
            return maybeJudge;
        }
        
        return -1;
    }
};